runtime/Java/src/org/antlr/v4/runtime/atn/ParserATNSimulator.java

```diff
@@ -1552,6 +1552,55 @@ protected void closure_(ATNConfig config,
         }
 
 		for (int i=0; i<p.getNumberOfTransitions(); i++) {
+			// This block implements first-edge elimination of ambiguous LR
+			// alternatives as part of dynamic disambiguation during prediction.
+			// See antlr/antlr4#1398.
+			if (i == 0 && p.getStateType() == ATNState.STAR_LOOP_ENTRY && ((StarLoopEntryState)p).isPrecedenceDecision) {
+				boolean suppress = !config.context.isEmpty();
+				int limit = config.context.size();
+				if (config.context.hasEmptyPath()) {
+					limit--;
+				}
+
+				// Require all return states to return back to the same rule
+				// that p is in.
```

Comment:
:memo: We are eliminating lookahead for one outgoing edge. This will affect *all* of the ATN configurations represented by the prediction context graph, so we need to verify that all of the configurations are affected by the ambiguity, hence why we verify the target of all return states.